### DISCUSSION TITLE
Is LQI device<-->router or device<-->coordinator?

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#15524] [General]
The LQI values which are shown, are these the link quality numbers to the first hop (=first router) or the total link from the end device to the coordinator?

In the latter case, how is this total link LQI calculated? For example, if I have the individual LQIs (device->router1, router1->router2, router2->coordinator), how would I obtain the total LQI?

**Comment by lux73** [comment]
https://github.com/Koenkk/zigbee2mqtt/discussions/6694 😉

**Reply by eigenphase** [reply]
Nice thank you!!

Sadly this answer makes me more confused: according to it, the LQI of the last hop is shown ... and last seems tk refer to uplink (end device to coordinator) based on the example @koenkk gave. But this neither makes sense nor does it fit my experience.

suppose I have coordinator and after 10m. Bulb1. Straight down the line Another Bulb2.
Then a bunch of end devices.

None of these end devices can reach the coordinator on its own so all of them will connect to Bulb2. Bulb2 will in turn route via Bulb1. Now the last hop is Coordinator->Bulb1. In other words it is the same for all of the end devices and hence would have same LQI. But this is definitely not what I am seeing
